services/api/src/authMiddleware.ts

```diff
@@ -33,108 +35,99 @@ const parseBearerToken = R.compose(
+      .send({ errors: [{ message: 'Unauthorized - Bearer token required' }] });
     return;
   }
 
-  req.authToken = token;
-
-  next();
-};
-
-const keycloak = async (
-  req: RequestWithAuthData,
-  res: Response,
-  next: NextFunction
-) => {
-  // Allow access to status without auth.
-  if (req.url === '/status') {
-    next();
-    return;
-  }
-
-  try {
-    const grant: any = await getGrantForKeycloakToken(req.authToken);
-
-    const ipAddress = getClientIp(req);
-    req.headers.ipAddress = ipAddress;
-
-    req.kauth = { grant };
-
-    const { azp: source, preferred_username, email } = grant.access_token.content;
-    const username = preferred_username ? preferred_username : 'unknown';
-
-    userActivityLogger.user_auth(`Keycloak authentication granted for '${username} (${email ? email : 'unknown'})' from '${source}'`,
-      { user: grant ? grant.access_token.content : null, headers: req.headers });
-
-  } catch (e) {
-    // It might be a legacy token, so continue on.
-    logger.debug(`Keycloak token auth failed: ${e.message}`);
-  }
-
-  next();
-};
-
-const legacy = async (
-  req: RequestWithAuthData,
-  res: Response,
-  next: NextFunction
-) => {
-  // Allow access to status without auth.
-  if (req.url === '/status') {
-    next();
-    return;
-  }
-
-  // Allow keycloak authenticated sessions
-  if (req.kauth) {
-    next();
+  // Fail early if the JWT data isn't valid
+  const decodedToken = decode(token, { json: true, complete: true });
+  if (decodedToken === null) {
+    res.status(401).send({
+      errors: [{ message: 'Unauthorized - Bearer token malformed' }]
+    });
     return;
   }
 
-  try {
-    const legacyCredentials = await getCredentialsForLegacyToken(req.authToken);
-    req.legacyCredentials = legacyCredentials;
-
-    const ipAddress = getClientIp(req);
-    req.headers.ipAddress = ipAddress;
-
-    const { sub, iss } = legacyCredentials;
-    const username = sub ? sub : 'unknown';
-    const source = iss ? iss : 'unknown';
-    userActivityLogger.user_auth(`Legacy authentication granted for '${username}' from '${source}'`, 
-      { user: legacyCredentials ? legacyCredentials : null, headers: req.headers });
+  req.authToken = token;
 
-    next();
-  } catch (e) {
-    res.status(403).send({
-      errors: [{ message: `Forbidden - Invalid Auth Token: ${e.message}` }]
+  if (isLegacyToken(decodedToken)) {
+    try {
+      const legacyCredentials = await getCredentialsForLegacyToken(token);
+      req.legacyCredentials = legacyCredentials;
+
+      const { sub, iss } = legacyCredentials;
+      const username = sub ? sub : 'unknown';
+      const source = iss ? iss : 'unknown';
+      userActivityLogger.user_auth(
+        `Legacy authentication granted for '${username}' from '${source}'`,
+        {
+          user: legacyCredentials ? legacyCredentials : null,
+          headers: req.headers
+        }
+      );
+
+      return next();
+    } catch (e) {
+      res.status(401).send({
```

Comment:
I changed the HTTP error code in a few places from `403 Forbidden` to `401 Unauthorized` when there was an error evaluating the supplied token, is that a BC break we care about or not?

Reply:
I don't think we need to care about this. 

For lagoon-cli, it will just capture all errors anyway so a change of 403 to 401 here is not a big issue for it. But things I don't know about too much being the UI, would it have an issue with this?

Reply:
Tested a bunch of stuff, couldn't find any errors in our tools